### ISSUE TITLE
3.8.3 causes calico-controllers to crash

### ISSUE THREAD

**Issue by djschny** [projectcalico/kube-controllers#420]
calico-contollers pod crashes constantly.

## Expected Behavior

to stay running

## Current Behavior

keeps crashing with no valid log output

## Possible Solution

Revert https://github.com/projectcalico/kube-controllers/milestone/92

## Steps to Reproduce (for bugs)

1. kubeadm cluster with version v1.14.4
2. install calico using `kubectl apply -f https://docs.projectcalico.org/v3.8/manifests/calico.yaml`
3. observe continual crashing controller pod
4.

## Your Environment

* Kubernetes version (`kubectl version`): 1.14.4
* Manifest file used to deploy k8s policy:
* Operating System and version: Ubuntu 16.04
* Link to your project (optional):

I turned on debug logging for the calico controller and we see the following:

```
log: exiting because of error: log: cannot create log: open /tmp/kube-controllers.calico-kube-controllers-6f684bbd9b-t2rnn.unknownuser.log.WARNING.20191016-203811.1: no such file or directory
```

Upon further investigation a new version of calico (v3.8.3) was released about 12 hours ago:

* https://github.com/projectcalico/kube-controllers/milestone/92
* https://hub.docker.com/r/calico/kube-controllers/tags?page=1&name=3.8

Apparently when we `kubectl apply -f https://docs.projectcalico.org/v3.8/manifests/calico.yaml` it automatically points to the latest bugfix version. So prior to today it was pulling v3.8.2 and they updated that URL to point to the latest 3.8.3 release:

https://github.com/projectcalico/kube-controllers/milestone/92?closed=1

Outside of this bug of the crashing, how can we lock to a specific point release version of the YAML so people can be protected from future changes?

**Comment by caseydavenport** [comment]
Thanks for reporting. Looking into this now.

To pin to a specific patch release, you can use the manifests within the release archive attached to the calico release.

For example for v3.8.2: https://github.com/projectcalico/calico/releases/tag/v3.8.2

**Comment by caseydavenport** [comment]
I believe this is the fix: https://github.com/projectcalico/kube-controllers/pull/422

**Comment by caseydavenport** [comment]
Merged - will release a v3.8.4 with the above fix this week.

**Comment by seh** [comment]
We've been mounting an `emptyDir` volume and mounting it at _/tmp_ to work around this problem.
